bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -46,6 +51,14 @@
   private Context operationContext;
   private Context rpcContext;
 
+  private RetryOptions retryOptions;
+
+  // The number of times we've retried after a timeout
+  private AtomicInteger timeoutRetryCount = null;
```

Comment:
This feels a bit weird, it doesn't seem like this class is generally thread safe, why use AtomicInteger?